### PR TITLE
Refactor aura and task process system, implement 10 cards in EXPERT1 cardset

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Implement ability
    - Choose One
    - Silence
    - Enrage
- Implement 10 cards in EXPERT1 cardset
    - Shadowstep (EX1_144)
    - Southsea Deckhand (CS2_146)
    - Tirion Fordring (EX1_383)
    - Wrath (EX1_154)
    - Ironbeak Owl (CS2_203)
    - Spiteful Smith (CS2_221)
    - Venture Co. Mercenary (CS2_227)
    - Blade Flurry (CS2_233)
    - Wild Pyromancer (NEW1_020)
    - Acolyte of Pain (EX1_007)
- Refactor `Aura` class
    - Change code for inheritance
    - Create `AdaptiveEffect` class
    - Create `EnrageEffect` class
- Refactor task process system
    - Create `TaskQueue` class
    - Add `ActivateTask()` method
    - Fix 'Tirion Fordring' (EX1_383) calls graveyard function twice when entity is destroyed (#266)
- Add utility function: `EraseIf()`
- Delete unused `DoBothPlayer` and `DoUntil` class

### CODE REVIEW

File: Includes/Rosetta/Enchants/Effects.hpp [hunk 1]
```diff
@@ -19,54 +19,55 @@ class Effects
 {
  public:
     //! Creates effect that increases attack by \p n.
-    static Effect AttackN(int n)
+    static Effect* AttackN(int n)
     {
-        return Effect(GameTag::ATK, EffectOperator::ADD, n);
+        return new Effect(GameTag::ATK, EffectOperator::ADD, n);
```
Comment (CreatorSeraph) [inline Includes/Rosetta/Enchants/Effects.hpp]: 반환값을 받아서 어디에선가 해제해주어야 하니
[[nodiscard]]를 추가해보는건 어떨까요?

Comment (utilForever) [reply Includes/Rosetta/Enchants/Effects.hpp]: 안그래도 코드 관리 툴에서 ``[[nodiscard]]``를 붙이자는 이야기를 했는데,
이 부분은 리팩토링 작업으로 따로 진행할게요. 이슈는 등록해 놓겠습니다.

File: Includes/Rosetta/Enums/TriggerEnums.hpp [hunk 2]
```diff
@@ -18,9 +18,13 @@ enum class TriggerType
     PLAY_CARD,   //!< The effect will be triggered when a player plays a card.
     CAST_SPELL,  //!< The effect will be triggered when a player plays a Spell
                  //!< card.
+    AFTER_CAST,  //!< The effect will be triggered after a spell is played.
     HEAL,        //!< The effect will be triggered when characters are healed.
     ATTACK,      //!< The effect will be triggered when characters attack.
     SUMMON,  //!< The effect will be triggered whenever a minion is summoned.
+    PREDAMAGE,  //!< Effect will be triggered when a character gets predamage.
+                //!< This event happens just before the character is actually
+                //!< damaged.
```
Comment (kozistr) [inline Includes/Rosetta/Enums/TriggerEnums.hpp]: please, align the indentation of the comments!

Comment (utilForever) [reply Includes/Rosetta/Enums/TriggerEnums.hpp]: Already aligned comment by clang-format.

Comment (kozistr) [reply Includes/Rosetta/Enums/TriggerEnums.hpp]: oh, i just mean, pad a `space` to the right!

Comment (utilForever) [reply Includes/Rosetta/Enums/TriggerEnums.hpp]: I don’t understand why I pad a space to the right. It is aligned using clang-format style.

Comment (kozistr) [reply Includes/Rosetta/Enums/TriggerEnums.hpp]: oh my mistake :( haha...

**Pull request merged**